src/bindgen/parser.rs

```diff
@@ -156,6 +156,25 @@ impl<'a> Parser<'a> {
         assert!(self.lib.is_some());
         self.parsed_crates.insert(pkg.name.clone());
 
+        // Before we do anything with this crate, we must first parse all of its dependencies.
+        // This is guaranteed to terminate because the crate-graph is acyclic (and even if it
+        // wasn't, we've already marked the crate as parsed in the line above).
+        if let Some(ref lib) = self.lib.as_ref() {
+            for (dep_pkg, cfg) in lib.dependencies(&pkg) {
+                if self.should_parse_dependency(&dep_pkg.name) {
+                    if let &Some(ref cfg) = &cfg {
+                        self.cfg_stack.push(cfg.clone());
+                    }
+
+                    self.parse_crate(&dep_pkg)?;
```

Comment:
I just tried using this branch and I got the following error from this part of the patch:

```
error[E0502]: cannot borrow `*self` as mutable because `self.lib` is also borrowed as immutable
   --> /Users/ham/.cargo/git/checkouts/cbindgen-108f22ebe155017c/a437c85/src/bindgen/parser.rs:169:21
    |
162 |         if let Some(ref lib) = self.lib.as_ref() {
    |                                -------- immutable borrow occurs here
...
169 |                     self.parse_crate(&dep_pkg)?;
    |                     ^^^^ mutable borrow occurs here
...
176 |         }
    |         - immutable borrow ends here
```

This is my Rust version:

```
$ rustc --version
rustc 1.35.0-nightly (96d700f1b 2019-04-10)
```